ts/src/bingx.ts

```diff
@@ -0,0 +1,2721 @@
+                    result['withdraw']['percentage'] = false;
+                }
+            }
+        }
+        return result;
+    }
+
+    async fetchDepositWithdrawFees (codes: string[] = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#fetchDepositWithdrawFees
+         * @description fetch deposit and withdraw fees
+         * @see https://bingx-api.github.io/docs/#/common/account-api.html#All%20Coins'%20Information
+         * @param {[string]|undefined} codes list of unified currency codes
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} a list of [fee structures]{@link https://docs.ccxt.com/en/latest/manual.html#fee-structure}
+         */
+        await this.loadMarkets ();
+        const response = await this.walletsV1PrivateGetCapitalConfigGetall (params);
+        const coins = this.safeValue (response, 'data');
+        return this.parseDepositWithdrawFees (coins, codes, 'coin');
+    }
+
+    async withdraw (code: string, amount, address, tag = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#withdraw
+         * @description make a withdrawal
+         * @see https://bingx-api.github.io/docs/#/common/account-api.html#Withdraw
+         * @param {string} code unified currency code
+         * @param {float} amount the amount to withdraw
+         * @param {string} address the address to withdraw to
+         * @param {string|undefined} tag
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @param {int} params.walletType 1 fund account, 2 standard account, 3 perpetual account
+         * @returns {object} a [transaction structure]{@link https://docs.ccxt.com/#/?id=transaction-structure}
+         */
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        let walletType = this.safeInteger (params, 'walletType');
+        if (walletType === undefined) {
+            walletType = 1;
+        }
+        if (!this.inArray (walletType, [ 1, 2, 3 ])) {
+            throw new BadRequest (this.id + ' withdraw() requires either 1 fund account, 2 standard futures account, 3 perpetual account for walletType');
+        }
+        const request = {
+            'coin': currency['id'],
+            'address': address,
+            'amount': this.numberToString (amount),
+            'walletType': walletType,
+        };
+        const network = this.safeStringUpper (params, 'network');
+        if (network !== undefined) {
+            request['network'] = network;
+        }
+        const response = await this.walletsV1PrivateGetCapitalWithdrawApply (this.extend (request, params));
+        const data = this.safeValue (response, 'data');
+        //
+        // TODO: Add response info
+        //
+        console.log (response);
+        console.log (data);
```

Comment:
@AlgoQ can you update this?

Reply:
Can you try out the withdrawal function first? Tried it out a couple of times but didn't seem to work (mentioned it on disc last week)

Reply:
@AlgoQ humm getting this error `bingx {"code":100500,"msg":"INTERNAL_SERVER_ERROR","timestamp":1689075404710}`

Reply:
@carlosmiei Yeah, same. Do we contact BingX?

Reply:
@AlgoQ I did that they said we should be using `POST` instead  of `GET`, can you try that?

Reply:
Oops ye stupid mistake, fixed in the last commit